api/src/database/channel-data.ts

```diff
@@ -10,12 +11,20 @@ const getIndex = (link: string, identityId: string) => `${link}-${identityId}`;
 export const getChannelData = async (
 	channelAddress: string,
 	identityId: string,
-	options: { limit?: number; index?: number; ascending: boolean },
+	options: ChannelLogRequestOptions,
 	secret: string
 ): Promise<ChannelData[]> => {
-	const { ascending, index, limit } = options;
+	const { ascending, index, limit, startDate, endDate } = options;
 
-	const query = { channelAddress, identityId };
+	let created: { $gte?: string; $lte?: string };
+	if (startDate && endDate) created = { $gte: startDate, $lte: endDate };
+	else if (startDate && !endDate) created = { $gte: startDate };
```

Comment:
General suggestion: Please always use {} although it is a online if statement and we remove this code